pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/store/SingleFileIndexDirectory.java

```diff
@@ -387,4 +401,53 @@ public boolean isIndexRemovalSupported() {
   public String toString() {
     return _segmentDirectory.toString() + "/" + _indexFile.toString();
   }
+
+  @VisibleForTesting
+  static List<IndexEntry> copyIndicesTo(File indexFile, Map<IndexKey, IndexEntry> toBeRetained, File tmpIdxFile)
+      throws IOException {
+    // Copy index from original index file and append to temp file.
+    // Keep track of the index entry pointing to the temp index file.
+    List<IndexEntry> retained = new ArrayList<>();
+    long nextOffset = 0;
+    // With FileChannel, we can seek to the data flexibly.
+    try (FileChannel srcCh = new RandomAccessFile(indexFile, "r").getChannel();
+        FileChannel dstCh = new RandomAccessFile(tmpIdxFile, "rw").getChannel()) {
+      for (Map.Entry<IndexKey, IndexEntry> next : toBeRetained.entrySet()) {
+        IndexKey key = next.getKey();
+        IndexEntry idxPos = next.getValue();
+        copyIndexTo(srcCh, idxPos.startOffset, idxPos.size, dstCh);
```

Comment:
I feel inlining it is more readable

Reply:
inlined next.getKey(), but will keep idxPos as it's referred 4 times here. idxPos as short for `index position`. lemme know how you'd like name it :)